src/serviceWorker/utils.ts

```diff
@@ -19,3 +19,17 @@ export function isDevelopment() {
 export function toURL(entry: string | PrecacheEntry): string {
   return typeof entry === 'string' ? entry : entry.url
 }
+
+type GroupedEntries = { onDemandCacheEntries: (string | PrecacheEntry)[]; precacheEntries: PrecacheEntry[] }
```

Comment:
I think it'll simplify readability if you return this as `mediaUrls: string[]`. It'll avoid further transforms, and let you name things knowing its acting on media.

```suggestion
type GroupedEntries = { mediaUrls: string[]; precacheEntries: PrecacheEntry[] }
```